src/cli/self_update.rs

```diff
@@ -280,6 +280,56 @@ impl SelfUpdateMode {
             Self::CheckOnly => "check-only",
         }
     }
+
+    /// Optionally performs a self-update: check policy, download, apply and exit.
```

Comment:
This ("move to a method") doesn't really make sense as a separate commit, since the function was added in this PR anyway. Just squash this?

Reply:
@djc Hmmm `self_update()` was an existing method. Otherwise I'd have squashed it since the beginning.